test/integration/controlplane/transformation/kms_transformation_test.go

```diff
@@ -481,34 +478,38 @@ resources:
 	}
 
 	// make sure cluster wide secrets read still works
-	_, err = test.restClient.CoreV1().Secrets("").List(context.TODO(), metav1.ListOptions{})
+	_, err = test.restClient.CoreV1().Secrets("").List(ctx, metav1.ListOptions{})
 	if err != nil {
 		t.Fatalf("failed to list secrets, err: %v", err)
 	}
 
 	// make sure cluster wide configmaps read still works
-	_, err = test.restClient.CoreV1().ConfigMaps("").List(context.TODO(), metav1.ListOptions{})
+	_, err = test.restClient.CoreV1().ConfigMaps("").List(ctx, metav1.ListOptions{})
 	if err != nil {
 		t.Fatalf("failed to list configmaps, err: %v", err)
 	}
 
 	// restart kube-apiserver with last applied encryption config and assert that server can start
-	previousConfigDir := test.configDir
-	test.shutdownAPIServer()
-	restarted = true
-	test, err = newTransformTest(t, "", true, previousConfigDir)
-	if err != nil {
-		t.Fatalf("failed to start KUBE API Server with encryptionConfig\n %s, error: %v", encryptionConfig, err)
+	if err = test.restartAPIServer(t, "", true); err != nil {
+		t.Fatalf("Failed to restart api server, error: %v", err)
 	}
-	defer test.cleanUp()
 
+	// confirm that reading a secret still works
```

Comment:
Yes this test failed for `Get` of a specific secrete before this restart apiserver fix because storage would be empty. And the previous tests only tested `List` and it would be ok as long as you are not looking for a secret that was created before the restart.